agoradatatools/process.py

```diff
@@ -136,27 +133,35 @@ def process_all_files(config_path: str = None, syn=None):
 
     load.create_temp_location(staging_path)
 
+    error_list = []
     if datasets:
         for dataset in datasets:
-            new_syn_tuple = process_dataset(
-                dataset_obj=dataset, staging_path=staging_path, syn=syn
-            )
-            # in the future we should log new_syn_tuples that are none
+            try:
+                process_dataset(dataset_obj=dataset, staging_path=staging_path, syn=syn)
+            except Exception as e:
+                print(e)
+                error_list.append({list(dataset.keys())[0]: e})
 
     destination = utils._find_config_by_name(config, "destination")
 
-    # create manifest
-    manifest_df = create_data_manifest(parent=destination, syn=syn)
-    manifest_path = load.df_to_csv(
-        df=manifest_df, staging_path=staging_path, filename="data_manifest.csv"
-    )
+    if not error_list:
+        # create manifest if there are no errors
+        manifest_df = create_data_manifest(parent=destination, syn=syn)
+        manifest_path = load.df_to_csv(
+            df=manifest_df, staging_path=staging_path, filename="data_manifest.csv"
+        )
 
-    load.load(
-        file_path=manifest_path,
-        provenance=manifest_df["id"].to_list(),
-        destination=destination,
-        syn=syn,
-    )
+        load.load(
+            file_path=manifest_path,
+            provenance=manifest_df["id"].to_list(),
+            destination=destination,
+            syn=syn,
+        )
+    else:
+        raise ADTDataProcessingError(
+            "Data Processing has failed for one or more data sources. Refer to the list of errors below to address issues:\n"
+            + str(error_list)
```

Comment:
Nit:  The string here could be pretty long. We could do something like to make it more readable. Do you have an output when there are multiple failures.

```
"\n".join(error_list)
```

Reply:
I tried this way initially, but `join()` only works when the members of the list are of type `str`. You get this error if you try to use it in this case:
```
TypeError                                 Traceback (most recent call last)
Cell In[8], line 1
----> 1 "\n".join(lst)

TypeError: sequence item 0: expected str instance, dict found
```

Reply:
I could do something like this if I change the error catching to be a list of strings though. Trying that now

Reply:
The latest commit changes the error log to a list of strings and makes the formatting a little nicer. example:
```
    raise ADTDataProcessingError(
errors.ADTDataProcessingError: 
Data Processing has failed for one or more data sources. Refer to the list of errors below to address issues:
genes_biodomains: 403 Client Error: Lack of READ permission on the parent entity.
neuropath_corr: 403 Client Error: Lack of READ permission on the parent entity.
proteomics: 403 Client Error: Lack of READ permission on the parent entity.
```